packages/cli/src/lib/package-managers.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  type Lockfile,
+  type PackageManager as Name,
+} from '@shopify/cli-kit/node/node-package-manager';
+
+export interface PackageManager {
+  name: Name;
+  lockfile: Lockfile;
+  installCommand: string;
+}
+
+export const packageManagers: PackageManager[] = [
+  {
+    name: 'npm',
+    lockfile: 'package-lock.json',
+    installCommand: 'npm ci',
+  },
+  {
+    name: 'yarn',
+    lockfile: 'yarn.lock',
+    installCommand: 'yarn install --immutable',
```

Comment:
I think `--immutable` is for newer versions of Yarn. The original v1 was the one introducing `--frozen-lockfile`. The newer ones also has this alias for backward compatibility.

```suggestion
    installCommand: 'yarn install --frozen-lockfile',
```

Reply:
But the [docs](https://yarnpkg.com/cli/install#:~:text=immutablePatterns%20configuration%20setting).-,For,-backward%20compatibility%20we) indicate the legacy `frozen-lockfile` is going to be removed in future versions, so not sure which one is best.